tests/components/animation.test.js

```diff
@@ -104,6 +104,27 @@ suite('animation', function () {
   });
```

Comment:
Here are some recommended changes for this PR commit.

1. The `setTimeout` function is used in the `remove` suite. This is not recommended because the timeout duration is arbitrary and may lead to flaky tests. Instead, you can use the `done` function in combination with event listeners to ensure that the tests only proceed once the necessary conditions have been met.

2. The `stopRelatedAnimations` test suite could benefit from some additional comments explaining the purpose and functionality of each test. This will make it easier for other developers to understand the intention of the tests and maintain them in the future.

3. In the `color animation` suite, the test `can animate color object directly` has a complex logic. It would be better to break this test down into smaller, more manageable tests. This would make it easier to identify specific issues if the test fails.

4. In the `basic animation` suite, the `handles non-truthy from value (i.e., 0)` test could benefit from a more descriptive test name. The current name does not clearly convey the purpose of the test.

5. In the `event emissions` suite, the `emits animationcomplete event twice` test could be simplified by using a counter variable instead of a boolean flag. This would make the test logic more straightforward and easier to understand.

6. In the `tick` suite, the test `only calls animejs animation.tick if playing` could benefit from a more descriptive test name. The current name does not clearly convey the purpose of the test.

7. The `vec3 animation` suite could benefit from some additional comments explaining the purpose and functionality of each test. This will make it easier for other developers to understand the intention of the tests and maintain them in the future.

Reply:
thanks for the comment, I did not touch most of the test suites you are mentioning and I am not sure I fill confident to modify them.
regarding the vec3 animation, I'll try to update with more comments if time but I did not touch aframe code since that PR more than a year ago so not sure I'll manage that soon